### PR TITLE
*: support show stats_healthy.

### PR DESCRIPTION
**Pull request by hanfei1991**
support command `show stats_healthy` to check if a table need to be analyze.

### CODE REVIEW

**Comment by hanfei1991** [comment]
@lamxTyler @winoros PTAL

File: executor/show_stats_test.go [hunk 1]
```diff
@@ -62,3 +63,23 @@ func (s *testSuite) TestShowStatsBuckets(c *C) {
...
+	do, _ := tidb.GetDomain(s.store)
+	do.StatsHandle().DumpStatsDeltaToKV()
+	tk.MustExec("analyze table t")
+	tk.MustQuery("show stats_healthy").Check(testkit.Rows("test t 100"))
+	tk.MustExec("insert into t values (3), (4), (5), (6), (7), (8), (9), (10)")
+	do.StatsHandle().DumpStatsDeltaToKV()
+	do.StatsHandle().Update(do.InfoSchema())
+	tk.MustQuery("show stats_healthy").Check(testkit.Rows("test t 19"))
+	tk.MustExec("analyze table t")
+	tk.MustQuery("show stats_healthy").Check(testkit.Rows("test t 100"))
```
Comment (winoros) [inline executor/show_stats_test.go]: add a `%`?

Comment (lamxTyler) [reply executor/show_stats_test.go]: But it is quite weird to have something like `-10%`?

File: executor/show_stats.go [hunk 2]
```diff
@@ -156,3 +156,21 @@ func (e *ShowExec) valueToString(value *types.Datum, size int) (string, error) {
...
+	h := do.StatsHandle()
+	dbs := do.InfoSchema().AllSchemas()
+	for _, db := range dbs {
+		for _, tbl := range db.Tables {
+			statsTbl := h.GetTableStats(tbl.ID)
+			if !statsTbl.Pseudo {
+				e.appendRow([]interface{}{
+					db.Name.O,
+					tbl.Name.O,
+					int64((1 - float64(statsTbl.ModifyCount)/float64(statsTbl.Count)) * 100.0),
```
Comment (coocood) [inline executor/show_stats.go]: I think the healthy should never be negative.

How about
```
(statsTabl.Count / (statsTbl.ModifyCount + statsTbl.Count)) * 100.0

```

Comment (coocood) [reply executor/show_stats.go]: Can we make sure that table count is not zero?

Comment (shenli) [reply executor/show_stats.go]: How about there is only 2 row in the table and the modify-cout is also 2? Will the health value be 0?

File: executor/show_stats_test.go [hunk 3]
```diff
@@ -62,3 +63,23 @@ func (s *testSuite) TestShowStatsBuckets(c *C) {
...
+}
```
Comment (coocood) [inline executor/show_stats_test.go]: Please add a case that all the rows have been deleted, then show stats_healthy.

File: executor/show_stats.go (hunk 2 above)
Comment (hanfei1991) [reply executor/show_stats.go]: If use `(statsTabl.Count / (statsTbl.ModifyCount + statsTbl.Count))`, the score is always too high. Say we havenot analyze but the score will be 50.
I think if modifycount is equal or larger than totol count, the score will be 0. That is better.

File: executor/show_stats.go [hunk 4]
```diff
@@ -156,3 +156,25 @@ func (e *ShowExec) valueToString(value *types.Datum, size int) (string, error) {
...
+func (e *ShowExec) fetchShowStatsHealthy() {
+	do := domain.GetDomain(e.ctx)
+	h := do.StatsHandle()
+	dbs := do.InfoSchema().AllSchemas()
+	for _, db := range dbs {
+		for _, tbl := range db.Tables {
+			statsTbl := h.GetTableStats(tbl.ID)
+			if !statsTbl.Pseudo {
+				var healthy int64
+				if statsTbl.ModifyCount < statsTbl.Count {
```
Comment (coocood) [inline executor/show_stats.go]: If table count is zero, the healthy is always 0 which is not.

File: executor/show_stats_test.go [hunk 5]
```diff
@@ -62,3 +63,28 @@ func (s *testSuite) TestShowStatsBuckets(c *C) {
...
+	tk.MustExec("insert into t values (3), (4), (5), (6), (7), (8), (9), (10)")
+	do.StatsHandle().DumpStatsDeltaToKV()
+	do.StatsHandle().Update(do.InfoSchema())
+	tk.MustQuery("show stats_healthy").Check(testkit.Rows("test t 19"))
+	tk.MustExec("analyze table t")
+	tk.MustQuery("show stats_healthy").Check(testkit.Rows("test t 100"))
+	tk.MustExec("delete from t")
+	do.StatsHandle().DumpStatsDeltaToKV()
+	do.StatsHandle().Update(do.InfoSchema())
+	tk.MustQuery("show stats_healthy").Check(testkit.Rows("test t 0"))
```
Comment (coocood) [inline executor/show_stats_test.go]: When the table count is zero, after analyze, the healthy should be 100%.

File: executor/show_stats.go [hunk 6]
```diff
@@ -156,3 +156,25 @@ func (e *ShowExec) valueToString(value *types.Datum, size int) (string, error) {
...
+					healthy = int64((1.0 - float64(statsTbl.ModifyCount)/float64(statsTbl.Count)) * 100.0)
```
Comment (coocood) [inline executor/show_stats.go]: For a frequently updated small table, the healthy can easily drop to zero, we should take that into account.

**Comment by coocood** [comment]
LGTM

File: executor/show_stats.go [hunk 7]
```diff
@@ -156,3 +156,27 @@ func (e *ShowExec) valueToString(value *types.Datum, size int) (string, error) {
...
 }
+
+func (e *ShowExec) fetchShowStatsHealthy() {
+	do := domain.GetDomain(e.ctx)
+	h := do.StatsHandle()
+	dbs := do.InfoSchema().AllSchemas()
+	for _, db := range dbs {
+		for _, tbl := range db.Tables {
+			statsTbl := h.GetTableStats(tbl.ID)
+			if !statsTbl.Pseudo {
```
Comment (shenli) [inline executor/show_stats.go]: Should we assign 0 or -1 for the table with statsTbl.Pseudo is true?

**Comment by shenli** [comment]
Please address the comment.

**Comment by hanfei1991** [comment]
@shenli 
there are too many system tables to show.

**Pull request merged**